openapi/nexus-internal.json

```diff
@@ -2138,6 +2138,19 @@
               }
             ]
           },
+          "cpus": {
+            "description": "Number of CPUs online on the sled.",
+            "type": "integer",
+            "format": "int64"
+          },
```

Comment:
I know we're trying to get the basics of provisioning going, but I think we don't really want to call this CPUs and we really want to be crisp about what we're tracking and why. Here are a few high-level thoughts with this as it applies to the PR as a whole.

We want to think in terms of the physical units: sockets, cores, and threads. While at the moment we're not performing core-aware scheduling, that is something that we know we need to do as a high-priority post-MVP item. That will change how we want to think about this and how we look at this. It also helps because the term `CPUs` is very ambiguous as to what it means. I think, even if we don't care about physical inventory right now that we want to think about this in terms of what this actually is, which is total threads in the system. If we'd rather call it `logical_cpus`, we could, though I think that term is equally vague and more OS-specific.

I think we likely want to think carefully about online. As what is going to be usable for guests will change, especially in the core aware world. Given that we're not going to be performing core-aware scheduling for a while and don't have a way to , I would consider just using the non-online value to start with. But wen we are, the online value is misleading for what a customer can use for VMs for two reasons:

* The moment a single thread in a core fails in a core-aware world, we can no longer perform VMM core-aware scheduling on that physical core.
* I think the more actionable thing for customers is understanding the total cores that are present, how many threads have failed, and what percent of the resources we're actually using for VMs are used.